apps/email/js/cards/base_card.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+define(function(require) {
+  var mozL10n = require('l10n!'),
+      base = require('./base');
+
+  // Set up the global time updates for all nodes.
+  (function() {
+    var formatter = new mozL10n.DateTimeFormat();
+    var updatePrettyDate = function updatePrettyDate() {
+      var labels = document.querySelectorAll('[data-time]');
+      var i = labels.length;
+      while (i--) {
+        labels[i].textContent = formatter.fromNow(
+          labels[i].dataset.time,
+          // the presence of the attribute is our indicator; not its value
+          'compactFormat' in labels[i].dataset);
+      }
+    };
+    var timer = setInterval(updatePrettyDate, 60 * 1000);
+
+    function updatePrettyDateOnEvent() {
+      clearTimeout(timer);
```

Comment:
(I know this is just moved code, but): We should change this clearTimeout to clearInterval for consistency, even though nsGlobalWindow::ClearTimeout and nsGlobalWindow::ClearInterval call the same underlying helper and the WHATWG spec seems to imply that they must behave the same because there seems to only be a single list of active timers.